src/dinitctl.cc

```diff
@@ -294,6 +294,11 @@ int dinitctl_main(int argc, char **argv)
         }
         else {
             // service name / other non-option
+            if (argv[i][0] == '\0') {
+                // Empty argument
+                cmdline_error = true;
```

Comment:
This is ok for now, but it would have been a good opportunity to give a more specific error message for this specific error.

Reply:
Is this OK?
```c
    if (empty_arg_error) {
        cerr << "dinitctl: Invalid empty argument in command line\n";
        cerr << "Try 'dinitctl --help' for more information and check the command line\n";
        return 1;
    }
```

Reply:
You don't need "in command line", or the 2nd line at all. Most of the errors are just a single line:

```
                        cerr << "dinitctl: --from should be followed by a service name" << std::endl;
                        return 1;
```

```
                    cerr << "dinitcheck: '--services-dir' (-d) requires an argument" << endl;
                    return 1;
```

```
                cerr << "dinitctl: unrecognized/invalid option: " << argv[i] << " (use --help for help)\n";
                return 1;
```

So be consistent with that!. You could put " (use --help for help)" at the end which would at least be consistent with that last example but I don't think that's necessary in this case, is the help really going to be helpful to the user for this error? - Think about it.

I know there is one case where it does say "Try 'dinitctl --help' for more information" - that is the generic error case where it is possible the user tried to use a command that isn't recognised for example (maybe they forgot the command or mis-spelled it. Then, seeing the help might be useful). But that is a very generic case. This is a specific error.

Reply:
So for example:

    cerr << "dinitctl: invalid empty argument\n"

... is fine.